pkg/git/gitclient/gitclient.go

```diff
@@ -268,11 +269,26 @@ func (g *GitClient) Branch(name string) error {
 
 func (g *GitClient) ValidateRemoteExists(ctx context.Context) error {
 	logger.V(3).Info("Validating git setup", "repoUrl", g.RepoUrl)
-	remote := g.Client.NewRemote(g.RepoUrl, gogit.DefaultRemoteName)
-	// Check if we are able to make a connection to the remote by attempting to list refs
-	_, err := g.Client.ListWithContext(ctx, remote, g.Auth)
+
+	isCodecommit, codeCommitUser, err := git.IsCodeCommitURL(g.RepoUrl)
 	if err != nil {
-		return fmt.Errorf("connecting with remote %v for repository: %v", gogit.DefaultRemoteName, err)
+		return err
+	}
+	if isCodecommit {
```

Comment:
this is still only validating the url format instead of making call to verify remote repo exists? can we add similar validation?

Reply:
verified that we can also use existing ListRemote for run time validation